build 1.1.0 package for Python 3.7

megies opened the pull request:
This PR is trying to package current stable version for Python 3.7

CC @krischer @ThomasLecocq

megies committed:
build 1.1.0 package for Python 3.7

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * Whenever possible python packages should use pip. See https://conda-forge.org/docs/meta.html#use-pip

megies committed:
try to add py37 to build matrices for all platforms

ocefpaf committed:
add py37 variable

ocefpaf commented:
@conda-forge-admin, please rerender

ocefpaf commented on conda-forge.yml:
```diff
@@ -4,3 +4,4 @@ travis:
 appveyor:
   secure:
     BINSTAR_TOKEN: tumuXLL8PU75WMnRDemRy02ruEq2RpNxeK3dz0MjFssnosPm2v4EFjfNB4PTotA1
+max_py_ver: '37'
```
@megies this is the "official way" to get py37 but bare in mind that the bot is doing that in topological order, so this PR may need to wait for packages upstream.

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.01

megies commented:
@ocefpaf Ok thanks, I was hoping to just use the setup that worked for the old build and add py37 on top, but it looks like there were massive changes. So yeah, I guess we might as well adapt to them right now..

ocefpaf commented on recipe/meta.yaml:
```diff
@@ -11,6 +11,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
+  skip: True  # [not py37]
```
You'll probably want to remove this and re-render again.

megies replied:
@ocefpaf For the current ObsPy version we have all packages built already besides the new py37. I just wanted to avoid having new redundant packages for all Python versions. But if you prefer I can just have all Py versions built in this PR regardless?

(I intended to remove that line for the next new ObsPy version build)

megies commented:
Hmm.. ok, I'll remove that restriction to Py37, I guess, since CI is not building anything right now seems like.

megies committed:
remove build restriction to py37 and bump build number

megies committed:
change install to use pip to make linter happy

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The `requirements/` sections should be defined in the following order: build, host, run; instead saw: host, build, run.

megies committed:
make linter happy

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

megies commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.01

megies commented:
Somehow the [recommended build command](https://conda-forge.org/docs/meta.html#use-pip) line [errors out with a jinja error on Travis](https://travis-ci.org/conda-forge/obspy-feedstock/jobs/437103084#L384). I'm tempted to switch back to the seemingly discouraged `python setup.py .. ` command that used to work?

megies committed:
revert a9b8639839e858d26a74126d6a1df323a00e6797

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * Whenever possible python packages should use pip. See https://conda-forge.org/docs/meta.html#use-pip

dsentinel commented:
Awesome!
I'm here for support

> Somehow the [recommended build command](https://conda-forge.org/docs/meta.html#use-pip) line [errors out with a jinja error on Travis](https://travis-ci.org/conda-forge/obspy-feedstock/jobs/437103084#L384). I'm tempted to switch back to the seemingly discouraged `python setup.py ..` command that used to work?

You can use the pip without the `{{ PYTHON }}` and just use `python`
I'm not sure why `PYTHON` isn't getting set. but you can still use pip

dsentinel commented:
`script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"`
to
`script: "python -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"`

megies committed:
try to fix packaging problems

kinda confused about the newly added "host" section complementing the
"build" section and that seems to be like pretty much the old "build"
section..??

megies committed:
add some meta info to recipy (extended description etc)

megies commented:
I think the current packaging problems have to do with the newly added `host` subsection (inside `requirements`) in the recipe.. It's pretty confusing to me, since how I read it pretty much everything that used to be in the `build` section now should go in the `host` section, even though there still is a `build` section that now should contain other stuff exclusively..

Maybe that also explains why it couldn't resolve `{{ PYTHON }}` earlier.. dunno

- https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#requirements-section
- https://travis-ci.org/conda-forge/obspy-feedstock/jobs/437134534#L3467

Let's see if these commits help..

@dsentinel you're very welcome to join the packaging fun.. you should have the rights to push to this branch on `obspy/obspy-feedstock` now (sent an invite)?

megies commented:
Ooof.. these were really big changes in conda-smithy it seems, keep running into more problems. Now the rerender created some numpy version combinations automatically that can't be resolved even during building. Dunno if we really have to fix sane numpy versions through this new `conda_build_config.yaml` mechanism: https://conda.io/docs/user-guide/tasks/build-packages/variants.html#creating-conda-build-variant-config-files

I'm stopping here for now @dsentinel feel free to pick up fixing the build.

megies committed:
remove pinned numpy versions

megies committed:
try to fix recipe

megies committed:
loosen tests for CI a bit: no CI fail on obspy runtests error reports

megies commented:
Test reports are a real mess, mostly due to some changes in how numpy prints (doctests).. but it looks OK enough to be bold and merge this one..

http://tests.obspy.org/?version=1.1.0&node=conda-forge

CC @lijunzh
